### ISSUE TITLE
'Error in sourced command file'

### ISSUE THREAD

**Issue by avncharlie** [Gallopsled/pwntools#1144]
(Apologies if this isn't the right place to ask this!)
I am a new user of pwntools, and am attempting to use pwntools gdb features.
However, when I try to run it, it says 'error in sourced command file, connection timed out'
Here is a screenshot:
<img width="1280" alt="screen shot 2018-04-30 at 6 42 59 pm" src="https://user-images.githubusercontent.com/24058782/39420117-6c4d244a-4ca6-11e8-87fb-1265c16a0cce.png">

Here is the template I used to run it:
```
#!/usr/bin/env python2

import pwn
pwn.context.terminal = ['tmux','splitw','-h']

# Debugger commands:
c = [
# 'b *main'
        ]

# Useful commands:
buf = pwn.cyclic(600)
# offset = pwn.cyclic_find('abcd')

cmd = '\n'.join(c)
p = pwn.gdb.debug(['./challenge',buf], cmd)

# If it's an interactive program: 
# p.sendline('abcd')
# data = p.recv()

p.interactive()
```
Thank you for your help.

**Comment by zachriggle** [comment]
Can you run with `python yourscript.py DEBUG` to show the actual GDB script which is emitted?

**Comment by avncharlie** [comment]
A screenshot of running it with `DEBUG`:
![image](https://user-images.githubusercontent.com/24058782/39460678-bb1d282a-4d48-11e8-9890-db8d0d9dd398.png)
The text from the left pane:
```
[+] Starting local process '/usr/bin/gdbserver' argv=['/usr/bin/gdbserver', '--multi', '--no-disable-randomization', 'localhost:0', './challenge', 'aaaabaaacaaadaaaeaaafaaagaaahaaaiaaajaaakaaalaaamaaanaaaoaaapaaaqaaaraaasaaataaauaaavaaawaaaxaaayaaazaabbaabcaabdaabeaabfaabgaabhaabiaabjaabkaablaabmaabnaaboaabpaabqaabraabsaabtaabuaabvaabwaabxaabyaabzaacbaaccaacdaaceaacfaacgaachaaciaacjaackaaclaacmaacnaacoaacpaacqaacraacsaactaacuaacvaacwaacxaacyaaczaadbaadcaaddaadeaadfaadgaadhaadiaadjaadkaadlaadmaadnaadoaadpaadqaadraadsaadtaaduaadvaadwaadxaadyaadzaaebaaecaaedaaeeaaefaaegaaehaaeiaaejaaekaaelaaemaaenaaeoaaepaaeqaaeraaesaaetaaeuaaevaaewaaexaaeyaaezaafbaafcaafdaafeaaffaafgaafhaafiaafjaafkaaflaafmaafnaafoaafpaafqaafraafsaaftaafuaafvaafwaafxaafyaaf'] : pid 1844
[DEBUG] Received 0x28 bytes:
    'Process ./challenge created; pid = 1848\n'
[DEBUG] Received 0x18 bytes:
    'Listening on port 39905\n'
[DEBUG] Wrote gdb script to '/tmp/pwntFc1QX.gdb'
    file ./challenge
    target remote 127.0.0.1:39905
    file ./challenge
[*] running in new terminal: /usr/bin/gdb -q  "./challenge" -x "/tmp/pwntFc1QX.gdb"
[DEBUG] Launching a new terminal: ['/usr/bin/tmux', 'splitw', '-h', '/usr/bin/gdb -q  "./challenge" -x "/tmp/pwntFc1QX.gdb"']
[*] Switching to interactive mode
$  
```
If it helps, I have GDB Peda installed (although it wasn't working without it either). Would system info help?

**Comment by zachriggle** [comment]
No idea, @avncharlie.  Works fine for me.  I expect your GDB is old and-or busted.  

* Can you run `./challenge` outside of Pwntools?
* Can you debug it outside of Pwntools?

https://asciinema.org/a/NxkPEICILqlTwHKV0YClMdzz3

**Comment by lxraa** [comment]
Have you solved this problem?

**Comment by lxraa** [comment]
Have you solved this problem?

**Comment by yangshouguo** [comment]
I have the same problem.
It seems that /tmp/pwncWzSaq.gdb file doesn't exist.
Still don't know how to solve this problem.

**Comment by robertlee22** [comment]
I got latest gdb and pwntools , but I still have this problem

**Comment by Arusekk** [comment]
This may be related to PEDA, please check if it works with PEDA disabled.

**Comment by sionnx** [comment]
yes，I‘m using peda，and I have this problem too.

**Comment by ssabogalp** [comment]
I have the same problem. Did anyone ever solved this?

**Comment by Arusekk** [comment]
You can probably try pwndbg instead of peda. Pwndbg is designed to have the look-and-feel of pwntools. This does not reproduce on my box, so I can't help much.

**Comment by J-jaeyoung** [comment]
I'm using pwndbg but have same problem...

**Comment by zachriggle** [comment]
It looks like the issue here may be that we're issuing the `file` command twice.  Per @avncharlie's post:

https://github.com/Gallopsled/pwntools/issues/1144#issuecomment-385594835

```
[DEBUG] Wrote gdb script to '/tmp/pwntFc1QX.gdb'
    file ./challenge
    target remote 127.0.0.1:39905
    file ./challenge
```

For those of you still having this issue (@J-jaeyoung) can you try updating Pwntools to the most recent beta version, and seeing if you still have this issue?

```sh
$ pip install pwntools==4.4.0b1
```

Thanks!

**Comment by nicolaipre** [comment]
If it helps anyone, I solved this problem by installing GDB 12.1 from source. It seems like gdb/gdbserver 9.2 did not support one of the i386 registers I needed to debug the specific file. 

```
./configure --prefix=/usr --with-system-readline --with-python=/usr/bin/python3
make
sudo make install
```
